test/libsolidity/Behaviour.h

```diff
@@ -0,0 +1,61 @@
+/*
+	This file is part of solidity.
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "Builtin.h"
+#include <memory>
+#include <vector>
+
+namespace solidity::frontend::test
+{
+
+class TestFunctionCall;
+
+class Behaviour
+{
+public:
+	Behaviour() = default;
+	virtual ~Behaviour() = default;
+
+	virtual void begin() {}
+	virtual void before(TestFunctionCall const& _call) { (void) _call; }
+	virtual void after(TestFunctionCall const& _call) { (void) _call; }
+	virtual void end() {}
+
+	virtual bool isValid(const TestFunctionCall& _call)
+	{
+		(void) _call;
+		return true;
+	}
+
+	virtual std::string format(
+		const TestFunctionCall& _call,
+		ErrorReporter& _errorReporter,
+		std::string const& _linePrefix,
+		bool const _renderResult,
+		bool const _highlight) const
+	{
+		(void) _call;
+		(void) _errorReporter;
+		(void) _linePrefix;
+		(void) _renderResult;
+		(void) _highlight;
```

Comment:
Just an idea, if you want to get rid of unused parameter, you can use `[[maybe-unused]]` in front of parameter when you define it. That way you don't need this `(void)` casts. I know it is noop, but still it looks a bit nicer with attribute instead

Reply:
True!